### ISSUE TITLE
Getting SIGSEGV calling IVRCompositor::Submit

### ISSUE THREAD

**Issue by mickvangelderen** [ValveSoftware/openvr#966]
The provided c++ OpenGL example from openvr works. For my own application I'm using the C API from rust. Initializeing, polling for events, querying the preferred device width and height for my framebuffer etc works. I get a SIGSEGV however when calling Submit. (only after adding a call to WaitGetPoses, without that the application was "unresponsive" from SteamVR's perspective).

I am at a loss, not sure what steps to take anymore. Any help or guidance is welcome. So far I've observed the following.

The SIGSEGV happens in clientvr.so:

```
frame #0: 0x00007ffff302ec67 vrclient.so`CVRSharedGLTexture::Update(void*, vr::VRTextureBounds_t*, CVRCompositorSharedTextures::CommandBuffer_t*) + 951
vrclient.so`CVRSharedGLTexture::Update:
->  0x7ffff302ec67 <+951>: movq   0x20(%r15), %r12
    0x7ffff302ec6b <+955>: testq  %r12, %r12
    0x7ffff302ec6e <+958>: movq   %rdx, 0x18(%r15)
    0x7ffff302ec72 <+962>: movq   %rax, 0x20(%r15)
```

I have never really disassembled anything but using IDA and lldb, it looks to me like `r15` is expected to be initialized as a pointer to a Semaphore: 

```
loc_B5E40:              ; this
mov     rdi, rbx
call    _ZN18CVRSharedGLTexture12GetSemaphoreEv ; CVRSharedGLTexture::GetSemaphore(void)
cmp     byte ptr [rbx+121h], 0
mov     r15, rax
jz      loc_B5A78
```

The code path that is taken through `CVRSharedGLTexture::Update` never actually reaches the above initialization of `r15` though. When `r15 + 0x20` is loaded, `r15` is `0`. 

`CVRSharedGLTexture::Update` does correctly query the width and height of the OpenGL texture I provide, having the expected dimensions 1512 x 1680 for the HTC VIVE. 

Project here: https://github.com/mickvangelderen/vr-lab/blob/master/src/main.rs

**Comment by mickvangelderen** [comment]
I've just discovered that unbinding the framebuffer color attachment textures before calling `submit` makes the SIGSEGV go away. If that is indeed the cause of the problem, I suspect the HTC Vive client implementation is most likely not very robust. It would be nice if this can be handled more elegantly.

For completeness sake: I was also trying to re-use the same framebuffer to render both eyes because I just wanted to get something on the HMD ASAP. Using two separate framebuffers with their own textures still produces the SIGSEGV if I bind the framebuffer color attachment textures before calling `submit`.

I have different problems now however. Opening a separate issue for that.
